sjsonnet/src/sjsonnet/Std.scala

```diff
@@ -508,8 +504,8 @@ class Std(private val additionalNativeFunctions: Map[String, Val.Builtin] = Map.
 
       def evalRhs(str: Val, ev: EvalScope, pos: Position): Val = {
         var s = str.asString
-        if (right) s = rightPattern.matcher(s).replaceAll("")
-        if (left) s = leftPattern.matcher(s).replaceAll("")
+        if (right) s = Platform.getPatternFromCache(rightPattern).matcher(s).replaceAll("")
+        if (left) s = Platform.getPatternFromCache(leftPattern).matcher(s).replaceAll("")
```

Comment:
Note that this is losing some of the benefits of the old specialization: I think it would be preferable to do a `Pattern.compile` on `leftPattern` and `rightPattern` above so that we can avoid the cache lookup cost.

Reply:
Good point